segments/cwd.py

```diff
@@ -21,17 +21,21 @@ def add_cwd_segment():
     if len(names) > max_depth:
         names = names[:2] + [u'\u2026'] + names[2 - max_depth:]
 
-    if not powerline.args.cwd_only:
-        for n in names[:-1]:
-            if n == '~' and Color.HOME_SPECIAL_DISPLAY:
-                powerline.append(' %s ' % n, Color.HOME_FG, Color.HOME_BG)
-            else:
-                powerline.append(' %s ' % n, Color.PATH_FG, Color.PATH_BG,
-                    powerline.separator_thin, Color.SEPARATOR_FG)
-
-    if names[-1] == '~' and Color.HOME_SPECIAL_DISPLAY:
-        powerline.append(' %s ' % names[-1], Color.HOME_FG, Color.HOME_BG)
+    if powerline.args.cwd_mode=='plain':
+        powerline.append(' %s%s ' % ('' if names[0] in '/~' else '/', '/'.join(names)), 
```

Comment:
I have been thinking about this a little bit. The way you have it works great -- but it will have issues on Windows. What I am thinking we might want to do is just use the `cwd` variable instead of joining the `names` variable. This will not convert `/home/user` to `~` though. I am ok with that as a first step though.

Ultimately, I think we need to do the following to properly solve that issue:

* Refactor the `get_short_path` into two separate functions. One that does string manipulation to replace `/home/user` with `~` and one that then breaks the string apart into `names`.
* In plain mode, just use the first function